include/dr/mhp/views/zip.hpp

```diff
@@ -193,11 +218,17 @@ class zip_iterator {
   }
 
 private:
-  // If it is not a remote iterator, assume it is a local iterator
-  auto static base_local(auto iter) { return iter; }
-
-  auto static base_local(dr::remote_iterator auto iter) {
-    return dr::ranges::local(iter);
+  auto static base_local(auto iter) {
+    if constexpr (dr::remote_iterator<decltype(iter)>) {
+      return dr::ranges::local(iter);
+    } else if constexpr (dr::localizable_range<decltype(*iter)>) {
```

Comment:
Still confused by all this.

Reply:
I tried to explain it in the answer of previous comment (see: _Not sure what this means. local zip interator is not contiguous..._) and refactored concept in https://github.com/oneapi-src/distributed-ranges/pull/278/commits/dacd410da4a86f36afe49d5bd73e57487e995303 to make it fitting into all of our concepts.